src/test/java/com/endava/internship/collections/StudentListTest.java

```diff
@@ -399,26 +417,188 @@ void subListWithInvalidIndexesCaseTwo(int firstIndex) {
+        Student vova = new Student(names[1],birthdays[1],details[1]);
+        Student petea = new Student(names[0],birthdays[0],details[0]);
+        StudentList studentList = new StudentList();
+        studentList.add(valik);
+        studentList.add(vova);
+        studentList.add(null);
+        studentList.add(petea);
+        studentList.add(null);
+        //when
+        boolean result= students.addAll(studentList);
+        //then
+        assertAll(
+                ()->assertTrue(result,"Returned true because of operation succeeds"),
+                ()->assertEquals(studentList.get(0),students.get(0),"The order should be like in sourse list"),
+                ()->assertEquals(studentList.get(1),students.get(1),"The order should be like in sourse list"),
+                ()->assertEquals(studentList.get(2),students.get(2),"The order should be like in sourse list")
+        );
     }
 
     @Test
-    void testToString() {
+    void containsAllFails() {
+        //given
+        Student valik = new Student(names[1],birthdays[2],details[3]);
+        Student vova = new Student(names[1],birthdays[1],details[1]);
+        Student petea = new Student(names[0],birthdays[0],details[0]);
+        students.add(valik);
+        students.add(vova);
+        StudentList studentList = new StudentList();
+        studentList.add(valik);
+        studentList.add(vova);
+        studentList.add(petea);
+        //when
+        boolean ifContainsAll= students.containsAll(studentList);
+        //then
+        assertFalse(ifContainsAll,"Because of students does not contains all StudentList,result is false");
     }
 
     @Test
-    void containsAll() {
+    void containsAllSucceeds() {
+        //given
+        Student valik = new Student(names[1],birthdays[2],details[3]);
+        Student vova = new Student(names[1],birthdays[1],details[1]);
+        Student petea = new Student(names[0],birthdays[0],details[0]);
+        students.add(valik);
+        students.add(vova);
+        StudentList studentList = new StudentList();
+        studentList.add(valik);
+        studentList.add(vova);
+        //when
+        boolean ifContainsAll= students.containsAll(studentList);
+        //then
+        assertTrue(ifContainsAll,"Because of students contains all StudentList,result is true");
+    }
+    @Test
+    void containsAllThrows() {
+        //given
+        Student valik = new Student(names[1],birthdays[2],details[3]);
+        Student vova = new Student(names[1],birthdays[1],details[1]);
+        students.add(valik);
+        students.add(vova);
+        StudentList studentList = null;
+        //when
+        IllegalArgumentException exception =
+                assertThrows(IllegalArgumentException.class,()->students.containsAll(studentList),
+                        "Exception should be thrown");
+        //then
+        assertEquals(exception.getMessage(),"Argument, passed to the method is null",
```

Comment:
keep order for assertions
1 - expected value
2 - actual value
3 - message

Reply:
Fixed!